frontend/tests/utils/helpers.ts

```diff
@@ -119,19 +129,32 @@ interface TryUntilConfigs {
   maxRetry?: number;
   page: Page;
   silent?: boolean;
+  timeoutMs?: number;
 }
 
+const RETRY_TIMEOUT_MS = 3 * 60 * 1000;
+
 export async function tryUntil(
   assert: () => void,
-  { maxRetry = 50, page, silent = false }: TryUntilConfigs
+  {
+    maxRetry = 50,
+    timeoutMs = RETRY_TIMEOUT_MS,
```

Comment:
enhancement - new optional `timeoutMs` for cases where you don't care how many retries to do and just want to retry until it hits the max timeout

CC: @atarashansky I think this is the use case you needed a while ago, yeah?